larray/core.py

```diff
@@ -2893,6 +4681,8 @@ def broadcast_with(self, other):
                 other_axes = AxisCollection(other_axes)
         if self.axes == other_axes:
             return self
+        # AD? -- | = union and & = intersection
+        # AD? -- Why not just | ?
```

Comment:
Good question. I don't remember.  Could be related to handling of length 1 axes or anonymous axes, which are tricky beasts. At least, I know there was a good reason for this, but I cannot tell what exactly. The trick is there might not be a good reason for this any more (it might work with the current code). Just test it and see if it breaks any of the unit tests.